.github/workflows/zephyr-hal.yml

```diff
@@ -51,6 +49,7 @@ jobs:
         with:
           # Specific Branch, tag, or SHA to checkout (uses default branch if empty)
           ref: 'develop'
+          token: ${{ secrets.ZEPHYRHAL }}
```

Comment:
@sihyung-maxim have you tried using `ssh-key` instead?  I recently had some success with it on the MSDKGen repo (see [here](https://github.com/Analog-Devices-MSDK/MSDKGen/blob/main/.github/workflows/package.yml)).

Just need to add the SSH key as a deployment key in the hal_adi repo settings.

Reply:
Does your SSH key secret hold the private key? If so, I didn't know you use it as a token for `actions/checkout`. We have an SSH secret (`secrets.MSDK_SSH_KEY`), but the issue right now is pushing files to a target repo outside the source repo where the action was triggered. I'm not sure if the target repo settings need to update its permissions to allow this or not...

Reply:
Yes, the secret has the private key (I generated a new one just in case it gets compromised somehow) and then I just pass that in as `ssh-key`.

As long as the key is added as a deploy key to both repos it should work.

<img width="870" alt="image" src="https://github.com/analogdevicesinc/msdk/assets/38844790/84995c98-9c5f-4266-a32a-4db23922dee4">